topics/dev/tutorials/debugging/tutorial.md

```diff
@@ -0,0 +1,730 @@
+
+> ### {% icon comment %} Python Linting
+> For backend consistency, we also have a Python linter. While we won't be running you through that exercise today, you may encounter a Python linting issue in the future. The way to diagnose and treat these is very similar to the way to do it for the front end.
+{: .comment}
+
+> ### {% icon hands_on %} Hands-on: Fixing client linting error
+> 
+> 1. __Finding the failing test on GitHub__
+> 
+>    > One of the failing tests on GitHub says "get_code_and_test". Clicking on the word "Details" assocated with this failure will open a brief view of the CircleCI Checks, with a list of tests that are run as part of that command. You'll see that the one that is failing is js_lint. And if you click that, you'll be navigated to the CicleCI app, where you can see much more information about the failures.
+>    > 
+>    > Here you can see the exact command that ran to lint our codebase; it's `cd client && yarn run eslint`.
+>    > 
+>    > We can also see that it has found 1 error, where that error is located, and a description of the problem.
+>    > 
+> 2. __Running the linter locally__
+>    > 
+>    > Alright, so we've seen the failure, but we want to verify it locally now. For this, you'll need to open up a terminal.
+>    > 
+>    > Make sure your virtual environment is activated. If not, activate it: `. .venv/bin/activate`
+>    > 
+>    > Then you've got two options. You can run the same exact command that CircleCI uses `cd client && yarn run eslint` or `make client-lint`, which does `cd client && yarn run eslint && yarn run prettier-check`. This will also make sure your code conforms to formatting standards.
+>    > 
+>    > After running that, you should see the same linting error from CircleCI; you should know where it is and how to fix it too!
+>    > 
+> 3. __Finding and fixing the issue__
+>    > 
+>    > > ### {% icon solution %} Solution
+>    > > In the file `client/src/components/RuleBuilder/RuleComponent`, there is an unused import. Simply delete the line that says `import Vue from "vue";` 
+>    > {: .solution }
+>    > 
+>    > If you think you've got it, try running the test locally again to be sure.
+>    > 
+> 4. __Finishing up__
+>    > 
+>    > Whoo-hoo! If you're here, you've identified the error, ran the appropriate command locally, found the problem, fixed it, confirmed that the command runs without errors, and are ready to commit your change. Write a useful commit message like `Fix a client linting issue in RuleComponent` or something similar, making sure it is clear what you've done.
+>    > (See [Galaxy's contributor's guide](https://github.com/galaxyproject/galaxy/blob/dev/CONTRIBUTING.md#how-to-contribute) for details on how to write a useful and properly formatted commit message.)
+>    > 
+>    > Awesome work - you now understand how to solve client linting issues!
+{: .hands_on}
+
```

Comment:
```suggestion
5. __Little Hint__
 You also can auto-fix your linting problems with `yarn run prettier` and `yarn eslint --fix`
```

Reply:
Added as a tip to the linting section. Thanks!